src/dodal/devices/aperturescatterguard.py

```diff
@@ -123,22 +123,23 @@ def _set_raw_unsafe(self, positions: ApertureFiveDimensionalLocation) -> AndStat
             operator.and_, [motor.set(pos) for motor, pos in zip(motors, positions)]
         )
 
-    def _get_closest_position_to_current(self) -> SingleAperturePosition:
+    def _get_current_aperture_position(self) -> SingleAperturePosition:
         """
-        Returns the closest valid position to current position within {TOLERANCE_STEPS}.
+        Returns the closest valid position to current position using readback values
+        for SMALL, MEDIUM, LARGE. ROBOT_LOAD position defined when mini aperture y <= ROBOT_LOAD_Y.
         If no position is found then raises InvalidApertureMove.
         """
         assert isinstance(self.aperture_positions, AperturePositions)
-        for aperture in self.aperture_positions.as_list():
-            aperture_in_tolerence = []
-            motors = self._get_motor_list()
-            for motor, test_position in zip(motors, list(aperture.location)):
-                current_position = motor.user_readback.get()
-                tolerance = self.TOLERANCE_STEPS * motor.motor_resolution.get()
-                diff = abs(current_position - test_position)
-                aperture_in_tolerence.append(diff <= tolerance)
-            if np.all(aperture_in_tolerence):
-                return aperture
+        current_ap_y = float(self.aperture.y.user_readback.get())
+
+        if int(self.aperture.large.get()) == 1:
+            return self.aperture_positions.LARGE
+        elif int(self.aperture.medium.get()) == 1:
+            return self.aperture_positions.MEDIUM
+        elif int(self.aperture.small.get()) == 1:
+            return self.aperture_positions.SMALL
+        elif current_ap_y <= self.ROBOT_LOAD_Y:
+            return self.aperture_positions.ROBOT_LOAD
```

Comment:
Should: I feel that the robot load position should still have some tolerance on it - probably the same `MRES` * some n steps as before - otherwise we'll still end up with similar issues where the state is unrecognised when the `current_ap_y` is just a hair short of the canonical `ROBOT_LOAD` position. Since this position is well out of the way of collisions we can probably increase the default number of steps to 5 or something.

Reply:
Sorry, I suggested this logic to @katesmith280. I'm happy with a tolerance or without given that it's just a less than at the moment (so has some tolerance built in). If you think tolerance improves it then that's fine by me

Reply:
I pushed a new change that decoupled the the robot load position (31.4mm) and the permitted tolerance (5 mm). So the aperture position value ROBOT_LOAD now has a bit more meaning, i.e. position plus tolerance.

Reply:
Oh, sorry. Yeah it's not critical, we could always address it later if it causes problems - I did see the `<=` but was just worried that that tolerance was only one-sided and won't work if the move to robot load stopped slightly before the position, so I do think it's more robust with that in.